src/protocol/reading.rs

```diff
@@ -14,14 +20,32 @@ impl Reading for InnerNode {
         Default::default()
     }
 
+    /// Terminates WebSocet packets, decodes and forwards [AlgoMessage] payload to synthetic node's inbound queue.
     async fn process_message(&self, source: SocketAddr, message: Self::Message) -> io::Result<()> {
-        info!(parent: self.node().span(), "got a message from {}: {:?}", source, message);
+        let span = self.node().span();
+        debug!(parent: span, "got a message from {}: {:?}", source, message);
+
+        // Only binary messages are expected.
+        if message.opcode() != websocket_codec::Opcode::Binary {
+            warn!(parent: span, "not a binary opcode");
+            return Ok(());
+        }
+
+        // TODO(Rqnsom): maybe mv this algocodec to our inner node, encapsulated in a RefCell
```

Comment:
why isn't this codec created in `Reading::codec`? we should probably make a composite codec here, with this one being the innermost one

Reply:
```
    type Message = websocket_codec::Message;
    type Codec = WebsocketCodec;
    fn codec(&self, _addr: SocketAddr, _side: ConnectionSide) -> Self::Codec {
        Default::default()
    }
```
Composite codec sounds interesting. I wasn't sure how to do it. I was afraid that changing anything here would break pea2pea inner node 🤔, but I haven't thought about it too much.

Reply:
Done